homeassistant/components/iron_os/coordinator.py

```diff
@@ -45,10 +47,32 @@ async def _async_update_data(self) -> LiveDataResponse:
             # coordinator refresh, only after the device has disconnected
             # the device info is refetched
             self.device_info = await self.device.get_device_info()
+            self.update_device_info()
             return await self.device.get_live_data()
 
-        except CommunicationError as e:
-            raise UpdateFailed("Cannot connect to device") from e
+        except CommunicationError:
+            _LOGGER.debug("Cannot connect to device", exc_info=True)
+            return self.data
+
+    def update_device_info(self) -> None:
+        """Update data in config entry for DeviceInfo."""
+
+        update_info = {}
+        if self.config_entry.data.get(CONF_SERIAL_NUMBER) is None:
+            update_info[CONF_SERIAL_NUMBER] = self.device_info.device_sn
+            update_info[CONF_DEVICE_ID] = self.device_info.device_id
+
+        if (
+            self.device_info.build
+            and self.device_info.build != self.config_entry.data.get(CONF_SW_VERSION)
+        ):
+            update_info[CONF_SW_VERSION] = self.device_info.build
```

Comment:
Why do we store the sw version in the entry data? We might as well just update the device directly for this. If the next time an entity is added without `sw_version` it will just use the old ones.

Reply:
Oh, I didn't know it remembers that and I was overwriting it with None. That makes some things easier ^^